### ISSUE TITLE
Istio + cloudflare proxy + AuthorizationPolicy

### ISSUE THREAD

**Issue by crrazyman** [istio/istio#30562]
**This is a question**

Hello,

I just can't figure out how can i set up Istio in order to restrict access from only a few IP addresses to some services.

I have a k8s cluster in Azure AKS. The ingressgateway is patched with
`"externalTrafficPolicy":"Local"`

Also, the domain i'm using is in Cloudflare and it is proxied so when i get a request, i get ingressgateway logs:

`[2021-02-01T17:20:36.635Z] "GET / HTTP/1.1" 403 - "-" 0 19 0 - "167.233.5.50,172.69.55.77" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.2 Safari/605.1.15" "bfe7eb7e-af3f-43fb-b7d8-7b0719ef483f" "example.com" "-" - - 10.10.2.172:8443 172.69.55.77:59828 example.com -`

**167.233.5.50** - is my real IP

How can i set up an AuthorizationPolicy in order to allow access from only my IP address?

Now i have this:

```
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: ingress-policy
spec:
  selector:
    matchLabels:
      app: istio-ingressgateway
  action: ALLOW
  rules:
  - from:
    - source:
        ipBlocks: ["167.233.5.48/29"]
```
But it's not working because istio sees Cloudflare's IP as source IPs.

And yes, i need it to be proxied by Cloudflare. If i disable CF proxy it works but i need it to be enabled.

**Comment by howardjohn** [comment]
This sounds like a cloudflare question, not an Istio one. I would look into what functionality cloudflare has to support this. Could be proxy protocol, XFF header, etc. Istio can only rely on information provided by cloudflare proxy

**Comment by crrazyman** [comment]
@howardjohn as far as i know, Cloudflare adds a HTTP header **CF-Connecting-IP**
https://support.cloudflare.com/hc/en-us/articles/200170786-Restoring-original-visitor-IPs-logging-visitor-IP-addresses

**Comment by davidmortiz** [comment]
@crrazyman Cloudflare by default adds an X-Forwarded-For header and additionally can supply a CF-Connecting-IP request which I believe is formatted in exactly the same was as the XFF header.  As far as I know Cloudflare only works at layer 7 and doesn't support ProxyProtocol

**Comment by howardjohn** [comment]
Istio will look at the X-Forwarded-For since 1.8+ I believe. What version are you on?

**Comment by howardjohn** [comment]
Have you tried 1.8+?

**Comment by howardjohn** [comment]
Assuming this is fixed in 1.8 since we haven't heard back. Thanks!
